transmogrifier/sources/xml/marc.py

```diff
@@ -827,6 +709,174 @@ def get_call_numbers(cls, source_record: Tag) -> list[str] | None:
+                    [
+                        contents_item.rstrip(" ./-")
+                        for contents_item in contents_value.split(" -- ")
+                    ]
+                )
+        return contents or None
+
+    @classmethod
+    def get_contributors(cls, source_record: Tag) -> list[timdex.Contributor] | None:
+        """Retrieve contributors using data from relevant MARC fields.
+
+        The method starts by creating a dictionary where the keys are
+        contributor names and the values are a set of unique 'kind' values
+        retrieved from the MARC record (subfield code 'e'). When the value
+        is an empty set, this means that subfield code 'e' was blank or
+        missing from the record.
+
+        Using the dictionary, the method will create model.Contributor
+        instances for every unique 'kind' value associated with a
+        contributor name. When the value (in the dictionary) is an empty
+        set, the created instance will set kind="Not specified".
+        """
+        contributors: list = []
+        contributors_dict = defaultdict(set)
+        contributor_marc_fields = [
+            {
+                "tag": "100",
+                "subfields": "abcq",
+            },
+            {
+                "tag": "110",
+                "subfields": "abc",
+            },
+            {
+                "tag": "111",
+                "subfields": "acdfgjq",
+            },
+            {
+                "tag": "700",
+                "subfields": "abcq",
+            },
+            {
+                "tag": "710",
+                "subfields": "abc",
+            },
+            {
+                "tag": "711",
+                "subfields": "acdfgjq",
+            },
+        ]
+
+        for contributor_marc_field in contributor_marc_fields:
+            for datafield in source_record.find_all(
+                "datafield", tag=contributor_marc_field["tag"]
+            ):
+                if contributor_name := (
+                    cls.create_subfield_value_string_from_datafield(
+                        datafield,
+                        contributor_marc_field["subfields"],
+                        " ",
+                    )
+                ):
+                    contributor_name = contributor_name.rstrip(" .,")
+                    contributor_kinds = cls.create_subfield_value_list_from_datafield(
+                        datafield, "e"
+                    )
+                    contributors_dict[contributor_name].update(contributor_kinds)
+
+        for name, kinds in contributors_dict.items():
+            if len(kinds) == 0:
+                contributors.append(timdex.Contributor(value=name, kind="Not specified"))
+            else:
+                contributors.extend(
+                    [
+                        timdex.Contributor(value=name, kind=kind.strip(" .,"))
+                        for kind in sorted(kinds, key=lambda k: k.lower())
```

Comment:
Chose to sort the "kind" types before `timdex.Contributor` instances are created so that, in the event that a record _is_ updated, the ordering of `contributors` in the TIMDEX record doesn't change every time (though I don't expect TIMDEX users to notice it). In any case, this allows us to avoid needing to use `set()` in the `get_contributor` unit tests.